Wikipedia/Code/UINavigationBarAppearance+Extensions.swift

```diff
@@ -0,0 +1,27 @@
+import Foundation
+import UIKit
+
+@objc extension UINavigationBarAppearance {
+    @objc static func appearanceForTheme(_ theme: Theme, style: WMFThemeableNavigationControllerStyle) -> UINavigationBarAppearance {
+        let appearance = UINavigationBarAppearance()
+        appearance.configureWithOpaqueBackground()
+        appearance.backgroundColor = theme.colors.chromeBackground;
+        appearance.titleTextAttributes = theme.navigationBarTitleTextAttributes
+        let backgroundImage: UIImage?
+        switch style {
+            
+        case .editor:
+            backgroundImage = theme.editorNavigationBarBackgroundImage
+        case .sheet:
+            backgroundImage = theme.sheetNavigationBarBackgroundImage
+        case .gallery:
```

Comment:
Does navigation bar need to be hidden in this case as well?

Reply:
From looking at the old obj c code.

Reply:
Awesome catch, that was an oversight. I added that logic back in where it was before in `WMFThemeableNavigationController`'s `applyTheme` method. I don't really love that lived here before - I think logically it probably belongs more somewhere that isn't theming - but I didn't want to go down a rabbit hole of changing too much at this point.